packages/plugin-compiler/src/main/kotlin/io/realm/compiler/RealmModelSyntheticPropertiesGeneration.kt

```diff
@@ -335,23 +356,39 @@ class RealmModelSyntheticPropertiesGeneration(private val pluginContext: IrPlugi
                                             symbol = type.symbol
                                         )
                                     )
-                                    // Collection type
+                                    // Collection type: remember to specify it correctly here - the
+                                    // type of the contents itself is specified as "type" above!
+                                    val collectionTypeSymbol = when (value.collectionType) {
+                                        CollectionType.NONE -> PROPERTY_COLLECTION_TYPE_NONE
+                                        CollectionType.LIST -> PROPERTY_COLLECTION_TYPE_LIST
+                                        CollectionType.SET -> error("Sets not ready yet.")
+                                        CollectionType.DICTIONARY -> error("Dictionaries not ready yet.")
+                                    }
                                     putValueArgument(
                                         arg++,
                                         IrGetEnumValueImpl(
                                             startOffset = UNDEFINED_OFFSET,
                                             endOffset = UNDEFINED_OFFSET,
                                             type = collectionType.defaultType,
-                                            symbol = collectionTypes.first { it.name == PROPERTY_COLLECTION_TYPE_NONE }.symbol
+                                            symbol = collectionTypes.first {
+                                                it.name == collectionTypeSymbol
+                                            }.symbol
                                         )
                                     )
                                     // Link target
                                     putValueArgument(
                                         arg++,
                                         if (type == objectType) {
-                                            irString(backingField.type.classifierOrFail.descriptor.name.identifier)
-                                        } else
+                                            when (collectionTypeSymbol) {
+                                                PROPERTY_COLLECTION_TYPE_NONE ->
+                                                    irString(backingField.type.classifierOrFail.descriptor.name.identifier)
+                                                PROPERTY_COLLECTION_TYPE_LIST ->
+                                                    irString(backingField.symbol.descriptor.type.arguments[0].toString())
```

Comment:
I am not sure what this covers. Are lists somehow expressed with a _link_?

Reply:
The previous logic would add the link parameter from `backingField.type.classifierOrFail.descriptor.name.identifier` which in the case of a `RealmList<Object>` field, would return `RealmList`. That is wrong, it should be `Object`.